palette/src/luma/relative_contrast.rs

```diff
@@ -0,0 +1,57 @@
+use from_f64;
+use FloatComponent;
+
+/// A trait for calculating relative contrast between two colors.
+///
+/// W3C's Web Content Accessibility Guidelines (WCAG) 2.1 suggest a method
+/// to calculate accessible contrast ratios of text and background colors
+/// for those with low vision or color deficiencies, and for contrast of user
+/// interface graphics. The possible range of ratios is from 1:1 to 21:1.
+/// There is a Success Criterion for Contrast (Minimum) and a Success Criterion
+/// for Contrast (Enhanced), SC 1.4.3 and SC 1.4.6 respectively, which are
+/// concerned with text and images of text. SC 1.4.11 is a Success Criterion for
+/// "non-text contrast" such as user interface components and other graphics.
+/// The relative contrast is calculated by `(L1 + 0.05) / (L2 + 0.05)`, where
+/// `L1` is the luminance of the brighter color and `L2` is the luminance of the
+/// darker color both in sRGB linear space. Higher contrast ratio is generally
+/// desireable.
+///
+/// For more details of the guidelines:
+///
+/// [Success Criterion 1.4.3 Contrast (Minimum) (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum)
+///
+/// [Success Criterion 1.4.6 Contrast (Enhanced) (Level AAA)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-enhanced)
+///
+/// [Success Criterion 1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)
+pub trait RelativeContrast {
+    /// The return type for the contrast ratio.
+    type Scalar: FloatComponent;
```

Comment:
I think we can tone this requirement down a bit. :slightly_smiling_face: All we need is `PartialOrd + FromF64`, as far as I can tell, and it doesn't _have_ to be useful as a color component. The rest is implementation details. Part of why I'm commenting on this is that I'm considering reducing the trait requirements further, if possible. `Float`, for example, implies a lot of additional mathematical properties!